app/controllers/podcast_metrics_controller.rb

```diff
@@ -1,112 +1,273 @@
-      #     .group(:country_code, :subdiv_code, "DATE_TRUNC('WEEK', day) AS day")
-      #     .order(Arel.sql("SUM(count) AS count DESC"))
-      #     .limit(10)
-      # @top_countries =
-      #   Rollups::DailyGeo
-      #     .where(podcast_id: @podcast.id)
-      #     .select(:country_code, "SUM(count) AS count")
-      #     .group(:country_code)
-      #     .order(Arel.sql("SUM(count) AS count DESC"))
-      #     .limit(10)
-      @agent_apps_query =
-        Rollups::DailyAgent
-          .where(podcast_id: @podcast.id)
-          .select("agent_name_id AS code", "SUM(count) AS count")
-          .group("agent_name_id AS code")
-          .order(Arel.sql("SUM(count) AS count DESC"))
-      @agent_types_query =
-        Rollups::DailyAgent
-          .where(podcast_id: @podcast.id)
-          .select("agent_type_id AS code", "SUM(count) AS count")
-          .group("agent_type_id AS code")
-          .order(Arel.sql("SUM(count) AS count DESC"))
-      @agent_os_query =
-        Rollups::DailyAgent
-          .where(podcast_id: @podcast.id)
-          .select("agent_os_id AS code", "SUM(count) AS count")
-          .group("agent_os_id AS code")
-          .order(Arel.sql("SUM(count) AS count DESC"))
-
-      @episode_rollups = episode_rollups(@episodes, @recent_downloads_by_episode, @alltime_downloads_by_episode)
-      @agent_apps = Kaminari.paginate_array(@agent_apps_query).page(params[:agent_apps]).per(10)
-      @agent_types = Kaminari.paginate_array(@agent_types_query).page(params[:agent_types]).per(10)
-      @agent_os = Kaminari.paginate_array(@agent_os_query).page(params[:agent_os]).per(10)
+    @episode_rollups = episode_rollups(@episodes, @episodes_recent, @episodes_alltime)
+
+    render partial: "episodes_card", locals: {
+      interval: @interval,
+      date_range: @date_range,
+      episodes: @episodes,
+      episode_rollups: @episode_rollups
+    }
+  end
+
+  def uniques
+    @selection = uniques_params[:uniques_selection]
+
+    if clickhouse_connected?
+      @uniques_rollups =
+        Rollups::DailyUnique
+          .where(podcast_id: @podcast.id, day: (@date_start..@date_end))
+          .select("DATE_TRUNC('#{@interval}', day) AS day, MAX(#{@selection}) AS #{@selection}")
+          .group("DATE_TRUNC('#{@interval}', day) AS day")
+          .order(Arel.sql("DATE_TRUNC('#{@interval}', day) ASC"))
+          .load_async
     end
+
+    @uniques = {
+      rollups: @uniques_rollups,
+      color: colors[0]
+    }
+
+    render partial: "uniques_card", locals: {
+      interval: @interval,
+      selection: @selection,
+      uniques: @uniques,
+      date_range: @date_range
+    }
+  end
+
+  def dropdays
+    @episodes =
+      @podcast.episodes
+        .published
+        .order(first_rss_published_at: :desc)
+        .paginate(params[:episode_dropdays], params[:per])
+
+    @dropdays = @episodes.map do |ep|
```

Comment:
since dropdays relies on a date range relative to the episode, i'm currently looping through the selected episodes and running the query for each one. i do think building subqueries and running them through a single query will improve efficiency and performance, but for a first pass iteration i believe this is sufficient and the current result is not too slow.